### DISCUSSION TITLE
Go to line, ignoring empty line.

### DISCUSSION THREAD

**Discussion by JoseConseco** [folke/flash.nvim#80] [Show and tell]
This will ignore empty lines, and cursor will go to begining of line (rather that to very start - 0 col) 

```
      {
        "gl",
        mode = { "n", "x", "o" },
        function()
          require("flash").jump {
            search = { mode = "search" },
            highlight = { label = { after = { 0, 0 } } },
            pattern = "^\\s*\\S", -- match non-whitespace at start plus any character (ignores empty lines)
          }
          vim.cmd [[normal! ^]]
        end,
      },

```
![image](https://github.com/folke/flash.nvim/assets/13521338/2419610d-eb41-4ab9-ac56-6be591a0b00a)

I'm just not sure why range is limited to only few lines up and down (probably some config flag?) . But other that that should work ok.

**Comment by folke** [comment]
Nice!

Some tweaks:
- set `max_length=0` so all labels will be used and skips won't be calculated
- `matches=false` to hide the whitespace match
- changed the pattern a bit so you can also jump to empty lines

```lua
require("flash").jump({
  search = { mode = "search", max_length = 0 },
  highlight = {
    label = { after = { 0, 0 } },
    matches = false,
  },
  pattern = "^\\s*\\S\\?", -- match non-whitespace at start plus any character (ignores empty lines)
})
```

**Comment by vegetablesalad** [comment]
I just wrote something similar, but I wanted to control for before/after cursor search. 
For some reason it is always matching both directions, am I using this incorrectly?
@folke or forward doesn't work for search mode ?

```lua
    {
      "fo",
      mode = { "n", "x", "o" },
      function()
        require("flash").jump({
          search = { forward = false, mode = "search", max_length = 0 },
          label = { after = { 0, 0 } },
          pattern = "^\\s*\\S",
        })
      end,
      desc = "Flash line UP",
    },
    {
      "fl",
      mode = { "n", "x", "o" },
      function()
        require("flash").jump({
          search = { forward = true, mode = "search", max_length = 0 },
          label = { after = { 0, 0 } },
          pattern = "^\\s*\\S",
        })
      end,
      desc = "Flash line DOWN",
    },
```
<img width="650" alt="image" src="https://github.com/user-attachments/assets/1dab6090-0744-4970-9652-608ffd9933e7">
